### PR TITLE
Update the action to use the search API to check for first PR

### PR DESCRIPTION
**Pull request by kartben**
It's someone's first opened PR when there is exactly 1 PR where creation date is <= this PR's creation date. 
It's someone's first merged PR when there is exactly 1 merged PR where close date is <= this PR's close date.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -52,9 +52,10 @@ async function run() {
         issue.number
       );
     } else {
-      firstContribution = await isFirstPull(
+      firstContribution = await isFirstOpenedOrMergedPR(
         client,
         sender,
+        `${context.repo.owner}/${context.repo.repo}`,
```
Comment (stephanosio) [inline src/main.ts]: Why is this necessary? This seems to go against the (albeit questionable) change you have made in https://github.com/zephyrproject-rtos/action-first-interaction/commit/5d55935c19f4b6f45eaa113304c50487f53353fb#diff-7934bf411fea192ad8cd69e0a12911648a2842cb0f2409a8fb67b41b7069d757L128.

File: src/main.ts [hunk 2]
```diff
@@ -130,56 +120,39 @@ async function isFirstIssue(
...
+  const {status: getPRStatus, data: pr} = await client.rest.pulls.get({
     ...github.context.repo,
-    per_page: 100,
-    page: page,
-    state: 'all'
+    pull_number: curPullNumber
   });
-
-  if (status !== 200) {
+
```
Comment (stephanosio) [inline src/main.ts]: Nit: leading spaces

File: src/main.ts [hunk 3]
```diff
@@ -130,56 +120,39 @@ async function isFirstIssue(
...
-
-  if (status !== 200) {
+  
+  if (getPRStatus !== 200) {
     throw new Error(`Received unexpected API status code ${status}`);
   }
 
-  if (pulls.length === 0) {
-    return true;
+  let query = `repo:${repo} type:pr author:${sender}`;
```
Comment (stephanosio) [inline src/main.ts]:
```suggestion
  let query = `repo:${github.context.repo.owner}/${github.context.repo.repo} type:pr author:${sender}`;
```

File: src/main.ts [hunk 4]
```diff
@@ -130,56 +120,39 @@ async function isFirstIssue(
...
+  if(closed) {
```
Comment (stephanosio) [inline src/main.ts]: Nit: style

```suggestion
  if (closed) {
```

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "first-interaction-action",
-  "version": "1.1.2-zephyr",
+  "version": "1.1.3-zephyr",
```
Comment (stephanosio) [inline package.json]: I am not sure how this versioning system is going to work when the upstream repository releases `1.1.2` and `1.1.3`.

If we are going to version the forked action based on the upstream version number, it should be distinguishable from the upstream version number (e.g. If the upstream version is `v1.1.1`, then something like `v1.1.1.1` or `v1.1.1-1` or `v1.1.1-zephyr-1` etc.)

Comment (kartben) [reply package.json]: moved to `1.1.1-zephyr-1`, and will use a `v1.1.1-zephyr-1` tag when merged
fwiw the upstream repo very much looks like abandonware at this point, unfortunately.

File: src/main.ts (hunk 1 above)
Comment (kartben) [reply src/main.ts]: Good call. 100% an oversight.

**Review by stephanosio — APPROVED** [review]
